### PR TITLE
Add Solver Trampoline Gas Benchmarking

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds solver trampoline gas benchmarking so we can make informed decisions about optimizations.

### Test Plan

```
% npm run -s bench
gas used: 16507
```

### CODE REVIEW

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia4.giphy.com%2Fmedia%2FFHYfX7OLdCmYeENZIA%2Fgiphy.gif&token=benchmark" data-gitmeme-token="benchmark"></a>

File: scripts/deploy.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { ethers } from "hardhat";
 
 async function main() {
   const SolverTrampoline = await ethers.getContractFactory("SolverTrampoline");
-  const solverTrampoline = await SolverTrampoline.deploy();
+  const solverTrampoline = await SolverTrampoline.deploy(ethers.constants.AddressZero);
```
Comment (fedgiac) [inline scripts/deploy.ts:5]: I imagine this change isn't related and the deploy process still needs work?

**Review by fedgiac — APPROVED** [review]
Nice!

File: scripts/deploy.ts (hunk 1 above)
Comment (nlordell) [reply scripts/deploy.ts:5]: No, this is not needed for this. My local linter was complaining about it so I fixed it.

**Pull request merged**